### PR TITLE
Fix /sync when we have no events

### PR DESCRIPTION
**Pull request by erikjohnston**
We used a since token of 0 to mean that no token was given. However, naffka
streams start at 0. This causes clients to get stuck spinning forever until an
event is sent.

This changes it so that we pass around pointers instead, with nil meaning a
since token wasn't given.

### CODE REVIEW

File: src/github.com/matrix-org/dendrite/syncapi/sync/request.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ type syncRequest struct {
 	userID        string
 	limit         int
 	timeout       time.Duration
-	since         types.StreamPosition
+	since         *types.StreamPosition
```
Comment (richvdh) [inline src/github.com/matrix-org/dendrite/syncapi/sync/request.go]: worth  having a comment to explain what `nil` means?

File: src/github.com/matrix-org/dendrite/syncapi/sync/request.go [hunk 2]
```diff
@@ -70,13 +70,16 @@ func getTimeout(timeoutMS string) time.Duration {
 	return time.Duration(i) * time.Millisecond
 }
 
-func getSyncStreamPosition(since string) (types.StreamPosition, error) {
+// getSyncStreamPosition tries to pass a string taken from the API to a stream
```
Comment (richvdh) [inline src/github.com/matrix-org/dendrite/syncapi/sync/request.go]: "a string" seems a bit vague. "a 'since' token" ?

File: src/github.com/matrix-org/dendrite/syncapi/sync/requestpool.go [hunk 3]
```diff
@@ -93,7 +93,7 @@ func (rp *RequestPool) OnIncomingSyncRequest(req *http.Request, device *authtype
 		case <-timer.C:
 			return util.JSONResponse{
 				Code: 200,
-				JSON: types.NewResponse(syncReq.since),
+				JSON: types.NewResponse(currPos),
```
Comment (richvdh) [inline src/github.com/matrix-org/dendrite/syncapi/sync/requestpool.go]: is this an unrelated fix? looks ok, just wondering.

Comment (erikjohnston) [reply src/github.com/matrix-org/dendrite/syncapi/sync/requestpool.go]: I otherwise would need to dereference `syncReq.since`, though in practice it probably doesn't really matter which one we chose─other than using `currPos` might be a bit faster.

File: src/github.com/matrix-org/dendrite/syncapi/sync/request.go [hunk 4]
```diff
@@ -70,13 +70,16 @@ func getTimeout(timeoutMS string) time.Duration {
 	return time.Duration(i) * time.Millisecond
 }
 
-func getSyncStreamPosition(since string) (types.StreamPosition, error) {
+// getSyncStreamPosition tries to pass a 'since' token taken from the API to a
```
Comment (richvdh) [inline src/github.com/matrix-org/dendrite/syncapi/sync/request.go]: s/pass/parse/
